cmake/Modules/BuildCommon.cmake

```diff
@@ -43,7 +43,7 @@ set(BUILD_SHARED_LIBS YES CACHE BOOL "Build shared libraries. Default: YES")
 # This option is added by include(CTest). We define it here to let people know
 # that this is a standard option.
 set(BUILD_TESTING ON CACHE BOOL "Build tests")
-set(BUILD_DOCS OFF CACHE BOOL "Build documentation with make doc")
+set(BUILD_DOCS "" CACHE STRING "Build documentation with make doc. Supported values: <unset>, external, internal. external docs hid '*-draft*' and '*-internal* documentation pages and directories when building documentation")
```

Comment:
if you are going to change this, there are other uses of `BUILD_DOCS` like in `conda/build.sh` that need to be updated.